Update version-compatibility-reference.md (#3951)
astronomer/docs: software/version-compatibility-reference.md

|   0.34.1 - 0.34.2   |                 |                 |                 |       ✔️        |       ✔️        |       ✔️        |       ✔️        |       ✔️        |                 |
|   0.35.0 -0.35.1    |                 |                 |                 |       ✔️        |       ✔️        |       ✔️        |       ✔️        |       ✔️        |       ✔️        |

> ### General Guidelines for Upgrading Kubernetes Versions

Reviewer: ```suggestion
### General recommendations for Kubernetes upgrades
```

> During the initial phase of upgrading the managed node groups, there shouldn't be any immediate impact on the Astronomer or Airflow components, assuming no workloads are running on the nodes being upgraded. Performing a controlled rollout restart of the worker nodes will help minimize disruptions. Ensure that you monitor the health of the new nodes and workloads before decommissioning the old nodes.

Reviewer: ```suggestion
If there are no workloads running on the nodes you want to upgrade, there won't be an immediate impact on the Astronomer or Airflow components during the initial phase of upgrading your managed node groups. To minimize disruptions, however, perform a controlled rollout restart of the worker nodes. During your controlled rollout, monitor the health of the new nodes and workloads before decommissioning the old nodes.
```

> Ensure all necessary backups are in place before beginning the upgrade process. Verify that all Astronomer and Airflow components are running as expected on the new nodes. Once the Astronomer version is compatible with the EKS/AKS Cluster, you should not need to change anything from the Astro end. The upgrade will require restarting the Kubelet on each node, which will cause the Astro/Airflow components to restart.

Reviewer: ```suggestion
Before beginning the upgrade process, ensure you have all necessary backups ready. After upgrading, verify that all Astronomer and Airflow components are running as expected on the new nodes. After the Astronomer version is compatible with the EKS/AKS Cluster, you don't need to change you Astronomer configurations or settings. However, the upgrade requires restarting the Kubelet on each node, which causes the Astro and Airflow components to also restart.
```
